app/src/main/java/com/example/myapplication/fragments/MyContactsFragment.kt

```diff
@@ -66,35 +70,67 @@ class MyContactsFragment : Fragment() {
         adapter = UsersAdapter(object : UserActionListener {
 
             override fun onDeleteUser(user: User) {
-                val listBeforeDeleted = viewModel.usersLiveData.value
+                val listBeforeDeletedContact = viewModel.usersLiveData.value
                 viewModel.deleteUser(user)
-                showRestoreUserMessage(listBeforeDeleted)
+                showRestoreUserMessage(listBeforeDeletedContact)
             }
 
             override fun onUserDetails(user: User) {
                 val action =
-                    MyContactsFragmentDirections.actionMyContactsFragmentToContactsProfileFragment(
+                    ViewPagerFragmentDirections.actionViewPagerFragmentToContactsProfileFragment(
                         user.photo,
                         user.name,
                         user.career
                     )
                 findNavController().navigate(action)
             }
+
+            override fun onSelectUser(user: User) {
+                viewModel.selectUser(user)
+
+                if (!viewModel.isAnyContactSelect()) {
+                    adapter.changeModeStatus(false)
+                    updateRecyclerViewMargin(DEFAULT_MARGIN)
+                    binding.bucket.visibility = View.GONE
+                }
+            }
+
+            override fun onMultiSelectModeActive() {
+                updateRecyclerViewMargin(MULTISELECT_MODE_MARGIN)
+                binding.bucket.visibility = View.VISIBLE
+
+                binding.bucket.setOnClickListener {
+                    adapter.changeModeStatus(false)
+                    viewModel.deleteSelectedContacts()
+                    updateRecyclerViewMargin(DEFAULT_MARGIN)
+                    binding.bucket.visibility = View.GONE
```

Comment:
might want to make an extension funcion for View to change visibility, not critical